stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/querychannels/DefaultChatEventHandlerTest.kt

```diff
@@ -0,0 +1,165 @@
+        )
+
+        result `should be equal to` EventHandlingResult.Skip
+    }
+
+    @Test
+    fun `When received NotificationRemovedFromChannelEvent and the channel is not present, it should be Skip`() {
+        val cid = randomString()
+        val channel = randomChannel(cid = cid)
+        val eventHandler = DefaultChatEventHandler(MutableStateFlow(emptyList()))
+
+        val event = randomNotificationRemovedFromChannelEvent(cid, channel)
+
+        val result = eventHandler.handleNotificationRemovedFromChannelEvent(
+            event,
+            Filters.neutral()
+        )
+
+        result `should be equal to` EventHandlingResult.Skip
+    }
+
+    @Test
+    fun `When received NotificationRemovedFromChannelEvent and the channel is present, it should be removed`() {
+        val cid = randomString()
+        val channel = randomChannel(cid = cid)
+        val eventHandler = DefaultChatEventHandler(MutableStateFlow(listOf(channel)))
+
+        val event = randomNotificationRemovedFromChannelEvent(cid, channel)
+
+        val result = eventHandler.handleNotificationRemovedFromChannelEvent(
+            event,
+            Filters.neutral()
+        )
+
+        result `should be equal to` EventHandlingResult.Remove(cid)
+    }
+
+    @Test
+    fun `When received NotificationMessageNewEvent and the channel is absent, it should be added`() {
+        val cid = randomString()
+        val channel = randomChannel(cid = cid)
+        val eventHandler = DefaultChatEventHandler(MutableStateFlow(emptyList()))
+
+        val event = randomNotificationMessageNewEvent(cid, channel)
+
+        val result = eventHandler.handleNotificationMessageNewEvent(
+            event,
+            Filters.neutral()
+        )
+
+        result `should be equal to` EventHandlingResult.Add(channel)
+    }
+
+    @Test
+    fun `When received NotificationMessageNewEvent and the channel is present, it should be skiped`() {
+        val cid = randomString()
+        val channel = randomChannel(cid = cid)
+        val eventHandler = DefaultChatEventHandler(MutableStateFlow(listOf(channel)))
+
+        val event = randomNotificationMessageNewEvent(cid, channel)
+
+        val result = eventHandler.handleNotificationMessageNewEvent(
+            event,
+            Filters.neutral()
+        )
+
+        result `should be equal to` EventHandlingResult.Skip
+    }
+
+    private fun randomNotificationAddedToChannelEvent(
+        cid: String = randomString(),
+        channel: Channel = randomChannel()
+    ): NotificationAddedToChannelEvent {
+        return NotificationAddedToChannelEvent(
+            type = randomString(),
+            createdAt = Date(),
+            cid = cid,
+            channelType = randomString(),
+            channelId = randomString(),
+            channel = channel,
+            totalUnreadCount = randomInt(),
+            unreadChannels = randomInt(),
+        )
+    }
+
+    private fun randomNotificationRemovedFromChannelEvent(
```

Comment:
We have `Mother` file with top-level creating functions that can be reshared across the test codebase. Would you mind to move these functions there?